dojo/tools/terrascan/parser.py

```diff
@@ -0,0 +1,65 @@
+import json
+import hashlib
+from dojo.models import Finding
+
+
+class TerrascanParser(object):
+    """
+    A class that can be used to parse the terrascan JSON report file
+    """
+
+    # table to match tfsec severity to DefectDojo severity
+    SEVERITY = {
+        "HIGH": "High",
+        "MEDIUM": "Medium",
+        "LOW": "Low",
+    }
+
+    def get_scan_types(self):
+        return ["Terrascan Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Terrascan Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON output for Terrascan scan report."
+
+    def get_findings(self, filename, test):
+        data = json.load(filename)
+        dupes = {}
+        if data.get('results').get('violations'):
+            for item in data.get('results').get('violations'):
```

Comment:
This conditions can fail if `results` attribute is not here.
```suggestion
        if 'results' not in data and 'violations' not in data.get('results'):
            raise ValueError("missing mandatory attribute 'results'")
        for item in data.get('results').get('violations'):
```